BlazarService/src/main/java/com/hubspot/blazar/listener/LaunchingRepositoryBuildVisitor.java

```diff
@@ -68,41 +70,55 @@ protected void visitLaunching(RepositoryBuild build) throws Exception {
     }
   }
 
-  private Set<Module> findModulesToBuild(RepositoryBuild build, Set<Module> modules) {
+  private Set<Module> findModulesToBuild(RepositoryBuild build, Set<Module> allModules) {
     CommitInfo commitInfo = build.getCommitInfo().get();
 
     final Set<Module> toBuild = new HashSet<>();
     if (build.getBuildTrigger().getType() == Type.MANUAL) {
-      toBuild.addAll(modules);
+      if (!build.getBuildOptions().isPresent() || build.getBuildOptions().get().getModuleIds().isEmpty()) {
+        toBuild.addAll(allModules);
+      } else {
+        final Set<Integer> requestedModuleIds = build.getBuildOptions().get().getModuleIds();
+        for (Module module : allModules) {
+          if (requestedModuleIds.contains(module.getId().get())) {
+            toBuild.add(module);
+          }
+        }
+        addDownstreamModules(build, allModules, toBuild);
```

Comment:
We should check if this build option is enabled before adding downstreams right?